core/src/main/java/org/infinispan/xsite/status/DefaultTakeOfflineManager.java

```diff
@@ -94,18 +97,38 @@ private static Optional<CacheConfigurationException> findConfigurationError(Thro
             Optional.empty();
    }
 
-   @Start
-   public void start() {
-      String localSiteName = rpcManager.getTransport().localSiteName();
-      config.sites().allBackupsStream()
+   @Inject
+   public void inject(EmbeddedCacheManager cacheManager, Configuration configuration, RpcManager rpcManager, TimeService timeService, EventLogManager eventLogManager, @ComponentName(KnownComponentNames.CACHE_NAME) String cacheName) {
```

Comment:
Why is the logic now in the `inject` method not `start`?

Reply:
Half of the fields are only used to populate the `offlineStatus` map. It is a waste to keep them and it does not make the code less readable (IMO). I reduced from 9 fields to 4 fields :smile:

Reply:
I need to replace the `RpcManager` with the `Transport`. 1sec

Reply:
Shouldn't the `inject` implementations only set fields though? From the docs:

```quote
 * Methods annotated with this Inject annotation should *only* set class fields. They should do nothing else.
 * If you need to do some work to prepare the component for use, do it in a {@link Start} method since this is only
 * called once when a component starts.
```

Reply:
I know I'm bending the rules to reduce the memory usage :)

Reply:
I get that, but are they any consequences to this/do we do it elsewhere? What stage of the cache lifecycle is `inject` called vs `start`?

Reply:
There are no consequences in doing this if you keep in mind the rules:

* Inject methods/fields can be invoked multiple times after the start. For example, if a test replaces the `TimeService`, the field is set again or the method annotated with `@Inject` is invoked.
* Components may not be started. It is safe here since they are all global components.